### PR TITLE
Bhr 19

### PR DESCRIPTION
**Pull request by RobertHebel**

### CODE REVIEW

File: frontend/validations/login.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    });
+  }
+
+  return {
+    errors,
+    isValid: isEmpty(errors)
+  };
+}
+
+
```
Comment (adam-golab) [inline frontend/validations/login.js]: empty lines at the end of file

File: frontend/webpack.config.dev.js [hunk 2]
```diff
@@ -41,6 +41,10 @@ module.exports = {
       inject: true,
     }),
   ],
+  node: {
```
Comment (adam-golab) [inline frontend/webpack.config.dev.js]: Why did you add this?

File: frontend/validations/login.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import isEmpty from 'lodash/isEmpty';
+import Joi from 'joi';
+
+export default function validateInput(data) {
+  let errors = {};
+
+  const schema = Joi.object().keys({
```
Comment (adam-golab) [inline frontend/validations/login.js]: You should move schemas to separate file. To reuse it in different places, for example in reset password form

File: frontend/webpack.config.dev.js (hunk 2 above)
Comment (RobertHebel) [inline frontend/webpack.config.dev.js]: I have added this because of the error "Module not found: Error: Can't resolve 'dns' (...)". I think it's related to jio library

File: frontend/validations/login.js (hunk 3 above)
Comment (RobertHebel) [inline frontend/validations/login.js]: done

File: frontend/webpack.config.dev.js (hunk 2 above)
Comment (adam-golab) [inline frontend/webpack.config.dev.js]: Could you add this to `webpack.config.prod.js` too?

File: frontend/src/containers/App.js [hunk 4]
```diff
@@ -9,6 +10,12 @@ class App extends Component {
   render() {
     return (
       <div>
+        <div className="nav">
```
Comment (adam-golab) [inline frontend/src/containers/App.js]: Navbar should be a separate component.

File: frontend/test/components/common/TextFieldGroup.test.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  it('check if all required fields are passed', () => {
+    const onChangeHandler = sinon.spy();
+    const wrapper = shallow(<TextFieldGroup
+                              field="test field"
+                              label="test label"
+                              value="text value"
+                              onChange={onChangeHandler}
+                            />);
+
+    expect(wrapper.instance().props.value).to.be.equal('text value');
```
Comment (adam-golab) [inline frontend/test/components/common/TextFieldGroup.test.js]: Here you should check if it's rendered. Not only passed as props

File: frontend/test/containers/Login.test.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    wrapper.instance().onChange( { target: { name: 'username', value: 'Jedrzej@koson.pl' } });
+    expect(wrapper.state('username')).to.be.equal('Jedrzej@koson.pl');
+  });
+
+  it('should not return any error when correct credentials are submitted', () => {
+    const onSubmitHandler = sinon.spy();
+    const wrapper = shallow(<Login onSubmit={onSubmitHandler}/>);
+    wrapper.instance().onChange({ target: { name: 'username', value: 'Jedrzej@koson.pl' } });
+    wrapper.instance().onChange({ target: { name: 'password', value: 'Alutka69' } });
+    wrapper.instance().onSubmit({ preventDefault: sinon.spy() });
```
Comment (adam-golab) [inline frontend/test/containers/Login.test.js]: What does this line do?

File: frontend/src/containers/Login.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import TextFieldGroup from '../components/common/TextFieldGroup';
+import validateInput from '../../validations/login';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as authActions from '../actions/authActions';
```
Comment (adam-golab) [inline frontend/src/containers/Login.js]: Could you change it to explicit imports? For example `import { something } from '../actions/authActions';` If there aren't any actions just remove this line

File: frontend/src/components/common/TextFieldGroup.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+const TextFieldGroup = ({ field, value, label, error, type, onChange, checkUserExists }) =>
+  <div className={classnames('form-group', { 'has-error': error } )}>
```
Comment (RobertHebel) [inline frontend/src/components/common/TextFieldGroup.js]: it will be fixed on branch bhr-20

File: frontend/src/components/common/TextFieldGroup.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    <label className="control-label">{label}</label>
```
Comment (RobertHebel) [inline frontend/src/components/common/TextFieldGroup.js]: fixed on branch bhr-20

File: frontend/test/validations/login.test.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import { expect } from 'chai';
+import validateInput from '../../validations/login';
+
+describe('Tests for input validation', () => {
+  it('should return username error when incorrect email is passed', () => {
+    const userData = {
+      username: 'AndrzejDuda.com',
```
Comment (RobertHebel) [inline frontend/test/validations/login.test.js]: changed to foo/bar/baz convention on branch bhr-20

File: frontend/src/components/common/TextFieldGroup.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    <input
```
Comment (RobertHebel) [inline frontend/src/components/common/TextFieldGroup.js]: I've migrated to redux-form on branch bhr-20 and bhr-21

File: frontend/validations/login.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import isEmpty from 'lodash/isEmpty';
```
Comment (RobertHebel) [inline frontend/validations/login.js]: I removed lodash from validation
